cli/util/uploader.go

```diff
@@ -0,0 +1,432 @@
+				if err != nil {
+					select {
+					case errCh <- fmt.Errorf("failed to process header for OID %s: %w", currOid.String(), err):
+					default:
+					}
+					return
+				}
+
+				index := uint(blockIndex) / indexFileSize
+				offset := (uint(blockIndex) - index*indexFileSize) * oidSize
+				res := make([]byte, oidSize)
+				currOid.Encode(res)
+				copy(buffer[offset:], res)
+				processedCounter.Add(1)
+			}
+		}()
+	}
+
+	for i := existingIndexCount; i < expectedIndexCount; i++ {
+		startIndex := i * indexFileSize
+		endIndex := startIndex + indexFileSize
+
+		for j := int(startIndex); j < int(endIndex); j += searchBatchSize {
+			remaining := int(endIndex) - j
+			end := j + min(searchBatchSize, remaining)
+			prm := client.PrmObjectSearch{}
+			filters := object.NewSearchFilters()
+			filters.AddFilter(blockAttributeKey, fmt.Sprintf("%d", j), object.MatchNumGE)
+			filters.AddFilter(blockAttributeKey, fmt.Sprintf("%d", end), object.MatchNumLE)
+			prm.SetFilters(filters)
+
+			objectIDs, err = neofs.ObjectSearch(ctx.Context, &p, account.PrivateKey(), containerID.String(), prm)
+			if err != nil {
+				return fmt.Errorf("no OIDs found for index files: %w", err)
+			}
+
+			for _, currOid := range objectIDs {
+				oidCh <- currOid
+			}
+		}
+
+		for processedCounter.Load() < int32(indexFileSize) {
+			select {
+			case err := <-errCh:
+				return err
+			default:
+				time.Sleep(100 * time.Millisecond)
+			}
+		}
+
+		processedCounter.Store(0)
+
+		select {
+		case err := <-errCh:
+			return err
+		default:
+		}
+		attrs := []object.Attribute{
+			*object.NewAttribute(attributeKey, strconv.Itoa(int(i))),
+			*object.NewAttribute("size", strconv.Itoa(int(indexFileSize))),
+			*object.NewAttribute("timestamp", strconv.FormatInt(time.Now().Unix(), 10)),
+			*object.NewAttribute("block-attribute", blockAttributeKey),
+		}
+		err = uploadObj(ctx.Context, p, user.NewAutoIDSignerRFC6979(account.PrivateKey().PrivateKey), account.PrivateKey().GetScriptHash(), containerID, buffer, attrs)
+		if err != nil {
+			return fmt.Errorf("failed to upload index file %d: %w", i, err)
+		}
+		fmt.Fprintf(ctx.App.Writer, "Uploaded index file %d\n", i)
+	}
+	close(oidCh)
+	return nil
+}
+
+// uploadObj uploads the block to the container using the pool.
+func uploadObj(ctx context.Context, p pool.Pool, signer user.Signer, owner util.Uint160, containerID cid.ID, objData []byte, attrs []object.Attribute) error {
+	var (
+		ownerID          user.ID
+		hdr              object.Object
+		ch               checksum.Checksum
+		v                = new(version.Version)
+		prmObjectPutInit client.PrmObjectPutInit
+	)
+
+	ownerID.SetScriptHash(owner)
+	hdr.SetPayload(objData)
+	hdr.SetPayloadSize(uint64(len(objData)))
+	hdr.SetContainerID(containerID)
+	hdr.SetOwnerID(&ownerID)
+	hdr.SetAttributes(attrs...)
+	hdr.SetCreationEpoch(1)
+	v.SetMajor(1)
+	hdr.SetVersion(v)
+
+	checksum.Calculate(&ch, checksum.TZ, objData)
+	hdr.SetPayloadHomomorphicHash(ch)
```

Comment:
We need to calculate/set it only if it's enabled in the network.

Reply:
```
(base) ekaterinapavlova@MacBook-Air-4 neofs-node % ./bin/neofs-cli netmap netinfo -r st1.storage.fs.neo.org:8080
Epoch: 26809
Network magic: [net 0x572dfa5] 91414437
Time per block: 15s
NeoFS network configuration (system)
  Audit fee: 50000000
  Storage price: 100000
  Container fee: 1000000000
  Container alias fee: 200
  EigenTrust alpha: 0.1
  Number of EigenTrust iterations: 4
  Epoch duration: 240
  Inner Ring candidate fee: 10000000000
  Maximum object size: 67108864
  Withdrawal fee: 100000000
  Homomorphic hashing disabled: false
  Maintenance mode allowed: true
NeoFS network configuration (other)

```
in testnet and mainnet it is enabled

Reply:
But we'll use the same code for other networks as well, this should be based on settings.